control-plane/subcommand/acl-init/command_test.go

```diff
@@ -210,3 +213,131 @@ func TestRun_PerformsConsulLogin(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "token created via login: {\"component\":\"foo\"}", tok.Description)
 }
+
+// TestRun_WithAclAuthMethodDefinedWritesConfigJsonWithTokenMatchingSinkFile
+// executes the consul login path and validates the token is written to
+// acl-config.json and matches the token written to sink file.
+func TestRun_WithAclAuthMethodDefinedWritesConfigJsonWithTokenMatchingSinkFile(t *testing.T) {
+	tokenFile := common.WriteTempFile(t, "")
+	bearerFile := common.WriteTempFile(t, test.ServiceAccountJWTToken)
+	t.Cleanup(func() {
+		os.Remove(tokenFile)
+	})
+
+	k8s := fake.NewSimpleClientset()
+
+	// Start Consul server with ACLs enabled and default deny policy.
+	masterToken := "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
+
+	server, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+		c.ACL.DefaultPolicy = "deny"
+		c.ACL.Tokens.InitialManagement = masterToken
+	})
+	require.NoError(t, err)
+	defer server.Stop()
+	server.WaitForLeader(t)
+	cfg := &api.Config{
+		Scheme:  "http",
+		Address: server.HTTPAddr,
+		Token:   masterToken,
+	}
+	consulClient, err := api.NewClient(cfg)
+	require.NoError(t, err)
+
+	// Set up the Component Auth Method, this pre-loads Consul with bindingrule,
+	// roles and an acl:write policy so we can issue an ACL.Login().
+	test.SetupK8sComponentAuthMethod(t, consulClient, "test-sa", "default")
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:              ui,
+		k8sClient:       k8s,
+		bearerTokenFile: bearerFile,
+	}
+
+	code := cmd.Run([]string{
+		"-token-sink-file", tokenFile,
+		"-acl-auth-method", componentAuthMethod,
+		"-component-name", "foo",
+		"-http-addr", fmt.Sprintf("%s://%s", cfg.Scheme, cfg.Address),
+		"-init-type", "client",
+		"-acl-dir", "/tmp",
```

Comment:
should we create a temp dir for this test? I'm thinking of potential test pollution in the future because we're always using the same dir and aclconfig path.